sui-execution/latest/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -68,19 +81,20 @@ mod checked {
         pub metrics: Arc<LimitsMetrics>,
         /// The MoveVM
         pub vm: &'vm MoveVM,
+        /// The LinkageView for this session
+        pub linkage_view: LinkageView<'state>,
+        pub native_extensions: NativeContextExtensions<'state>,
         /// The global state, used for resolving packages
         pub state_view: &'state dyn ExecutionState,
         /// A shared transaction context, contains transaction digest information and manages the
         /// creation of new object IDs
         pub tx_context: &'a mut TxContext,
         /// The gas charger used for metering
         pub gas_charger: &'a mut GasCharger,
-        /// The session used for interacting with Move types and calls
-        pub session: Session<'state, 'vm, LinkageView<'state>>,
         /// Additional transfers not from the Move runtime
         additional_transfers: Vec<(/* new owner */ SuiAddress, ObjectValue)>,
         /// Newly published packages
-        new_packages: Vec<Object>,
+        new_packages: LinkedHashMap<ObjectID, MovePackage>,
```

Comment:
I think you would get away just fine with a `BTreeMap`, since it all get's thrown into a `BTreeMap` at the end. But it feels nice to preserve the order if we can I suppose

Reply:
The reason we need the order here is so that we can "unpublish".

Reply:
Why do you need the order for that?

Reply:
Although yes, you can do this just by remembering the package ID that you want to unpublish and passing that in!